igvc_perception/src/multiclass_segmentation/train.py

```diff
@@ -148,26 +153,67 @@
 test_data = SegmentationDataset(test_images_path, test_masks_path)
 infer_loader = DataLoader(test_data, batch_size=12, shuffle=False, num_workers=4)
 
-# Generates predictions on test data
-predictions = runner.predict_loader(
-    model=model,
-    loader=infer_loader,
-    resume=f"content/full_model2/checkpoints/best.pth",
-    verbose=False,
+# Get model predictions on test dataset
+predictions = np.vstack(
+    list(
+        map(
+            lambda x: x["logits"].cpu().numpy(),
+            runner.predict_loader(
+                loader=infer_loader, resume=f"content/full_model2/checkpoints/best.pth"
+            ),
+        )
+    )
 )
 
-# Show an image from the test dataset
-show_image = np.asarray(test_data[30]["image"])
-show_image = np.swapaxes(show_image, 2, 0)
-show_image = np.swapaxes(show_image, 1, 0)
-show_image = show_image.astype(np.uint8)
-np.shape(show_image)
-plt.imshow(show_image)
-
-# Show model prediction for image
-show_image = np.asarray(predictions[30])
-show_image = np.swapaxes(show_image, 2, 0)
-show_image = np.swapaxes(show_image, 1, 0)
-show_image = show_image.astype(np.float64)
-np.shape(show_image)
-plt.imshow(show_image)
+# Display sample prediction results
+pred_results = {}
+rand_nums = np.random.randint(low=1, high=148, size=18)
+rand_nums = np.insert(rand_nums, 0, 30)
+rand_nums = np.insert(rand_nums, 0, 141)
```

Comment:
What are these magic numbers?

Reply:
There's a lot of blank images inside of the dataset. Those images specifically show both lines and barrels being detected and are the best indicators of a well-trained model. I can add a comment explaining that.